sycl/source/detail/kernel_impl.cpp

```diff
@@ -44,16 +44,6 @@ kernel_impl::kernel_impl(sycl::detail::pi::PiKernel Kernel,
       MCreatedFromSource(IsCreatedFromSource),
       MKernelBundleImpl(std::move(KernelBundleImpl)),
       MKernelArgMaskPtr{ArgMask} {
-
-  sycl::detail::pi::PiContext Context = nullptr;
-  // Using the plugin from the passed ContextImpl
-  getPlugin()->call<PiApiKind::piKernelGetInfo>(
-      MKernel, PI_KERNEL_INFO_CONTEXT, sizeof(Context), &Context, nullptr);
-  if (ContextImpl->getHandleRef() != Context)
-    throw sycl::invalid_parameter_error(
-        "Input context must be the same as the context of cl_kernel",
-        PI_ERROR_INVALID_CONTEXT);
```

Comment:
This had to be moved up the chain to `kernel` constructor, to correctly throw the exception.